packages/ui-link/src/Link/index.tsx

```diff
@@ -37,226 +44,261 @@ import {
-      color,
-      href,
-      margin,
-      renderIcon,
-      iconPlacement,
-      isWithinText,
-      ...props
-    } = this.props
+    const renderIconElement = () => {
+      warn(
+        // if display prop is used, warn about icon or TruncateText
+        display === undefined,
+        '[Link] Using the display property with an icon may cause layout issues.'
+      )
+
+      // Map Link sizes to Lucide icon semantic size tokens
+      const linkSizeToIconSize = {
+        small: 'xs',
+        medium: 'sm',
+        large: 'md'
+      } as const
+
+      const iconSize = linkSizeToIconSize[size || 'medium']
+
+      // Lucide icons - pass size prop to control icon size
+      // Wrap with span for gap/spacing but without fontSize override
+      const isLucideIcon =
+        renderIcon &&
+        isValidElement(renderIcon) &&
+        (renderIcon as any).type?.displayName?.startsWith('wrapLucideIcon')
+
+      if (isLucideIcon && isValidElement(renderIcon)) {
+        // Lucide icons - clone the element with size prop
+        return (
+          <span css={styles?.icon}>
+            {cloneElement(renderIcon, { size: iconSize } as any)}
+          </span>
+        )
+      }
 
-    const { interaction } = this
+      // Non-Lucide icons or functions - render without size prop
+      return <span css={styles?.icon}>{callRenderProp(renderIcon as any)}</span>
+    }
 
     const isDisabled = interaction === 'disabled'
 
     const type =
-      this.element === 'button' || this.element === 'input'
-        ? 'button'
-        : undefined
-
-    const tabIndex = this.role === 'button' && !isDisabled ? 0 : undefined
-
-    return (
-      <View
-        {...passthroughProps(props)}
-        elementRef={this.handleElementRef}
-        as={this.element}
-        display={this.display}
-        margin={margin}
-        href={href}
-        onMouseEnter={onMouseEnter}
-        onClick={this.handleClick}
-        onFocus={this.handleFocus}
-        onBlur={this.handleBlur}
-        aria-disabled={isDisabled ? 'true' : undefined}
-        role={this.role}
-        type={type}
-        tabIndex={tabIndex}
-        css={this.props.styles?.link}
-        data-cid={combineDataCid('Link', this.props)}
-      >
-        {renderIcon && iconPlacement === 'start' ? this.renderIcon() : null}
-        {children}
-        {renderIcon && iconPlacement === 'end' ? this.renderIcon() : null}
-      </View>
+      element === 'button' || element === 'input' ? 'button' : undefined
+
+    // Determine tabIndex based on role and disabled state
+    const tabIndex = useMemo(() => {
+      if (isDisabled) {
+        // Disabled links should not be focusable
+        return -1
+      }
+      if (roleValue === 'button') {
+        // Elements with button role need explicit tabIndex
+        return 0
+      }
+      // Otherwise let the browser handle default focusability
+      return undefined
+    }, [isDisabled, roleValue])
+
+    return jsx(
```

Comment:
Please dont use this syntax (`jsx(`), we never use it in our codebase. What was your intention here?